internal/httputil/httpapi.go

```diff
@@ -47,9 +53,19 @@ type BasicAuth struct {
 	Password string `yaml:"password"`
 }
 
+type Consent bool
+
+const (
+	ConsentRequired    Consent = true
+	ConsentNotRequired Consent = false
+)
+
 // MakeAuthAPI turns a util.JSONRequestHandler function into an http.Handler which authenticates the request.
 func MakeAuthAPI(
-	metricsName string, userAPI userapi.UserInternalAPI,
+	metricsName string,
+	userAPI userapi.UserInternalAPI,
+	userConsentCfg config.UserConsentOptions,
+	requireConsent Consent,
```

Comment:
I would be very tempted to use functional options here so we don't need to constantly provide these 2 options.